src/Store.ts

```diff
@@ -0,0 +1,581 @@
+		//		in which their __proto__ will be set to Object.prototype. To provide
+		//		data model support, the returned object needs to be an instance of the model.
+		//		This can be accomplished by setting __proto__ to the model's prototype
+		//		or by creating a new instance of the model, and copying the properties to it.
+		//		Also, model's can provide their own restore method that will allow for
+		//		custom model-defined behavior. However, one should be aware that copying
+		//		properties is a slower operation than prototype assignment.
+		//		The restore process is designed to be distinct from the create process
+		//		so their is a clear delineation between new objects and restored objects.
+		// object: Object
+		//		The raw object with the properties that need to be defined on the new
+		//		model instance
+		// mutateAllowed: boolean
+		//		This indicates if restore is allowed to mutate the original object
+		//		(by setting its __proto__). If this isn't true, than the restore should
+		//		copy the object to a new object with the correct type.
+		// returns: Object
+		//		An instance of the store model, with all the properties that were defined
+		//		on object. This may or may not be the same object that was passed in.
+		const Model = this.Model;
+		if (Model && object) {
+			const prototype = Model.prototype;
+			const restore = prototype._restore;
+			if (restore) {
+				// the prototype provides its own restore method
+				object = restore.call(object, Model, mutateAllowed);
+			} else if (hasProto && mutateAllowed) {
+				// the fast easy way
+				// http://jsperf.com/setting-the-prototype
+				object['__proto__'] = prototype;
+			} else {
+				// create a new object with the correct prototype
+				object = <Hash<any>> lang.create(prototype, object, Model);
+			}
+		}
+		return object;
+	}
+
+	protected _setIdentity(object:{ [index: string]: any; set: any}, identityArg: any) {
+		// summary:
+		//		Sets an object's identity
+		// description:
+		//		This method sets an object's identity and is useful to override to support
+		//		multi-key identities and object's whose properties are not stored directly on the object.
+		// object: Object
+		//		The target object
+		// identityArg:
+		//		The argument used to set the identity
+		if (object.set) {
+			object.set(this.idProperty, identityArg);
+		} else {
+			object[ this.idProperty ] = identityArg;
+		}
+	}
+
+	create(properties: {}) {
+		// summary:
+		//		This creates a new instance from the store's model.
+		//	properties:
+		//		The properties that are passed to the model constructor to
+		//		be copied onto the new instance. Note, that should only be called
+		//		when new objects are being created, not when existing objects
+		//		are being restored from storage.
+		return  new this.Model(properties);
+	}
+
+	fetchRange(args: dstore.FetchRangeArgs): dstore.FetchPromise<T> {
+		throw new Error("This method is abstract;")
```

Comment:
I assumed that defining `fetchRange` as a method would be preferable, as if it were a property then my understanding is that when an inheriting class like `Request` overrides it, it would be getting set each time a `Request` store is constructed instead of on the prototype. I figure we might want to change a lot of the other 'abstract' members to be methods as well later on, but for now left them the way they are, because I haven't converted any classes that override them and been forced to make that decision.

And obviously what was implicit in that is that a method has to have an implementation provided where it is declared, necessitating throwing an error.